src/main.py

```diff
@@ -89,15 +95,47 @@ def start(url: str, map_slug: str, schema: str, style: str, wd: pathlib.Path) ->
 
     dataset_sheet = wb["datasetMetadata"]
     rows = dataset_sheet.iter_rows()
-    # always skip first row
-    next(rows)
-    header = [cell.value.strip() for cell in next(rows) if cell.value]
+
+    # header 1 = first row, filled with previous value if empty
+    h1, last_value = [], None
+    h1 = [
+        (last_value := cell.value.strip() if cell.value else last_value)
+        for cell in next(rows)
+    ]
+    # header 2 = second row, should not contain empty values
+    h2 = [cell.value.strip() for cell in next(rows) if cell.value]
+
+    display_metadata = {h1[0]: {}}
+    logging.info(f"h1: {display_metadata}")
+
     for row in rows:
-        row = [cell.value.strip() if isinstance(cell.value, str) else cell.value for cell in row]
+        row = [
+            cell.value.strip() if isinstance(cell.value, str) else cell.value
+            for cell in row
+        ]
         if not any(row):
             continue
-        dataset_metadata = dict(zip(header, row, strict=True))
-
+        # create display_metadata = {h1: {h2: value}}
+        dataset_metadata = dict(zip(h2, row, strict=True))
+        dataset_metadata = {
+            k: (v if v is not None else "") for k, v in dataset_metadata.items()
+        }
+        # Created nested dict with h1 as key for h2 and value
+        for h1_key, h2_key, value in zip(h1, h2, row):
+            if h1_key not in display_metadata:
+                display_metadata[h1_key] = {}
+            if value is not None:
+                display_metadata[h1_key][h2_key] = value
+            else:
+                display_metadata[h1_key][h2_key] = ""
+
+        # rename keys, convert None values to empty string, start each word with lower key
+        display_metadata = {k.replace(" ", "_"): v for k, v in display_metadata.items()}
+        display_metadata = {
+            k[0].lower() + k[1:]: v for k, v in display_metadata.items()
```

Comment:
instead of duplicating the statement, I would suggest creating a function that manipulates the key:
```python
def str_to_snake_case(text):
    return text.lstrip()[0].lower() + text.strip()[1:].replace(" ", "_")

display_metadata = {str_to_snake_case(k): v for k, v in display_metadata.items()}
```

Reply:
Also, this approach only converts first level keys, it would be better to convert the keys directly while populating `display_metadata`.

`h1` is used only for metadata, so it can be pre-converted to snake_case
```python
(last_value := str_to_snake_case(cell.value) if cell.value else last_value)
```

`h2` can be duplicated to a snake_case version outside of the loop:
```python
h2_snake_case = map(str_to_snake_case, h2)
```
Now we can zip already formatted keys
```python
for h1_key, h2_key, value in zip(h1, h2_snake_case, row):
...
```

Reply:
I added a str_to_snake_case function. But I let it handle multiple conversions, so that we only use snake_case in the template. Ok?